python-api/livy/client.py

```diff
@@ -0,0 +1,167 @@
+#
+# Licensed to Cloudera, Inc. under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  Cloudera, Inc. licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from ConfigParser import SafeConfigParser
+from StringIO import StringIO
+import os
+import codecs
+
+
+class HttpClient(object):
+    """A http based client for submitting Spark-based jobs to a Livy backend."""
+
+    __CONFIG_SECTION = 'env'
+    __LIVY_CLIENT_CONF_DIR = "LIVY_CLIENT_CONF_DIR"
+
+    def __init__(self, uri, load_defaults=True, conf_dict = dict()):
+        """
+         Loads the provided config and creates a http-based client
+         :param uri: Livy server uri generated from urlparse lib
+         :type uri: ParseResult generated from urlparse lib
+
+         >>> from urlparse import urlparse
+         >>> uri = urlparse("http://test")
+
+         :param boolean load_defaults: Default value is true. If true, loads the
+             livy client configuration from the files 'livy-client.conf' and
+             'spark-defaults.conf'. The path to the files should be defined in the
+             'LIVY_CLIENT_CONF_DIR' environment variable
```

Comment:
For me, it's confusing to have some options come from function arguments and others come from environmental variables. I personally prefer function arguments to environment variables, so that all options are visible in the function signature.